Automattic/woocommerce-payments
# UPE hover styling

Fixes #1829 

#### Changes proposed in this Pull Request

Hover states for tabs are generated following these rules:

* Tabs on hover will have a darker background than current background.
* If current background color has brightness under 50 (Storefront is 51), the tab on Hover becomes lighter.
* Color for labels or fillable SVGs do not change on Hover. If the combination new background color/ text color is not readable, we use a fallback of black or white, depending on the background color. (This is handled using Tinycolor library)










#### Testing instructions





* Enter your WC Options table and add the feature flag `_wcpay_feature_upe` with a value of 1. Stripe UPE payment gateway should now replace the Credit Card and be enabled at checkout.
* Please go to Payments > Settings and ensure that WooCommerce Payments is enabled.
* Add an item from your shop to your cart and proceed through a regular checkout. At checkout, select "WooCommerce Payments" and you should see the UPE appear below the payment gateway.
* Currently there is only one payment method active, cards, so not tabs are selected. To see the tabs, add the payments to [this array](https://github.com/Automattic/woocommerce-payments/blob/634d88056b48fd18f81787ea03bb597f24f3987f/includes/payment-methods/class-upe-payment-gateway.php#L110): [ 'bancontact', 'card', 'ideal' ]

Check that when you hover over the tabs, the backgrounds are darker. 
If the tab is very dark (Ex: 2021 theme) the background will be lighter.
![StorefrontHover](https://user-images.githubusercontent.com/8002881/119893680-4812db80-bf01-11eb-9a92-f5e7a6c1fcaa.gif)


-------------------

- [x] Tested on mobile (or does not apply) -> Tested but not comprehensively.
